lib/tests/streamlit/secrets_test.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2018-2021 Streamlit Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""st.beta_secrets unit tests."""
+import os
+import unittest
+from unittest.mock import patch, mock_open
+
+from toml import TomlDecodeError
+
+import streamlit as st
+from streamlit.secrets import _SECRETS_LOCATION
+
+MOCK_TOML = """
+# Everything in this section will be available as an environment variable
+db_username="Jane"
+db_password="12345qwerty"
+
+# Sub-sections are not loaded into os.environ
+[subsection]
+email="eng@streamlit.io"
+"""
+
+
+class SecretsTest(unittest.TestCase):
+    def setUp(self) -> None:
+        # st.beta_secrets modifies os.environ, so we save it here and
+        # restore in tearDown
+        self._prev_environ = dict(os.environ)
+
+    def tearDown(self) -> None:
+        st.beta_secrets._reset()
+        os.environ.clear()
+        os.environ.update(self._prev_environ)
+
+    @patch("builtins.open", new_callable=mock_open, read_data=MOCK_TOML)
+    def test_access_secrets(self, _):
+        self.assertEqual(st.beta_secrets["db_username"], "Jane")
+        self.assertEqual(st.beta_secrets["subsection"]["email"], "eng@streamlit.io")
+
+    @patch("builtins.open", new_callable=mock_open, read_data=MOCK_TOML)
+    def test_os_environ(self, _):
+        """os.environ gets patched when we load our secrets.toml"""
+        # We haven't loaded secrets yet
+        self.assertEqual(os.environ.get("db_username"), None)
+
+        # Load secrets
+        _ = st.beta_secrets["db_username"]
```

Comment:
This is more of a concern on the product side (cc @astrojams1): it feels weird to me that environment variables only get patched when the toml file is parsed but we don't have an `init()` or `load()` API in the secrets module.

It seems like a user that would prefer to use patched environment variables to access their secrets are discouraged from doing so since they'll have to do something to trigger a secrets file load, at which point you might as well just call `st.beta_secrets["foo"]`

Reply:
Thanks for pointing this out, and I agree! I'll resolve offline w/@astrojams1